addons/project_todo/models/res_users.py

```diff
@@ -30,7 +30,7 @@ def systray_get_activities(self):
                         END AS states
                      FROM mail_activity AS act
                      JOIN project_task AS t ON act.res_id = t.id
-                    WHERE act.res_model = 'project.task' AND act.user_id = %(user_id)s
+                    WHERE act.res_model = 'project.task' AND act.user_id = %(user_id)s AND act.active = true
```

Comment:
Your fix is clearly good but I think we could use `active_test` inside the context to know if we just need the active or also the archived ones.

```suggestion
                    WHERE act.res_model = 'project.task' AND act.user_id = %(user_id)s AND act.active in (TRUE, %(active)s)
```
and then we pass active_test as parameter like this:
```py
        self.env.cr.execute(query, {
            'user_id': self.env.uid,
            'active': self._context.get('active_test', True),
        })
```

We did that here:  <https://github.com/odoo/odoo/blob/9db9d9c512e2a879f590397d46bcdfb50a42fdeb/addons/project/models/project_task.py#L1691-L1700>